src/main/resources/templates/company.html

```diff
@@ -3,73 +3,85 @@
-                <input type="text" id="company-office" name="office" required th:field="*{office}">
+                    <label for="company-office">Company Office:</label>
+                    <input type="text" id="company-office" name="office" th:field="*{office}">
 
-                <label for="company-contactNo">Company Contact Number:</label>
-                <input type="text" id="company-contactNo" name="contactNo" required th:field="*{contactNo}">
+                    <label for="company-contactNo">Company Contact Number:</label>
+                    <span class="text-danger" th:if="${#fields.hasErrors('contactNo')}"
+                          th:errors="${companyForm.contactNo}">Error</span>
+                    <input type="text" id="company-contactNo" name="contactNo" th:field="${companyForm.contactNo}">
 
-                <input type="submit" value="Add Company">
-            </form>
-        </div>
+                    <input type="submit" value="Save Company">
+                </form>
+            </div>
 
+            <div class="company-list">
+                <h2>Company List</h2>
+                <div class="table-responsive-md">
+                    <table class="table table-striped table-bordered">
+                        <thead>
 
+                        <tr>
+                            <td>Company Name</td>
+                            <td>Contact Person</td>
+                            <td>Office Location</td>
+                            <td>Contact Number</td>
+                            <td colspan="2">Credentials</td>
+                        </tr>
+                        </thead>
+                        <tbody>
+                        <tr th:each="company : ${companyPage}">
+                            <td th:text="${company.companyName}"></td>
+                            <td th:text="${company.contactPerson}"></td>
+                            <td th:text="${company.office}"></td>
+                            <td th:text="${company.contactNo}"></td>
+                            <td>
+                                <form th:action="@{/company/edit}" method="get">
+                                    <input type="hidden" name="id" th:value="${company.id}"/>
+                                    <button type="submit" class="btn btn-primary">Edit</button>
+                                </form>
+                            </td>
+                            <td>
+                                <form th:action="@{/company/delete}" method="get">
+                                    <input type="hidden" name="id" th:value="${company.id}"/>
+                                    <button type="submit" class="btn btn-danger">Delete</button>
+                                </form>
+                            </td>
+                        </tr>
+                        </tbody>
+                    </table>
+                </div>
+                <span th:replace="paging.html"></span>
```

Comment:
That's not a parameterized fragment.
See https://www.thymeleaf.org/doc/tutorials/2.1/usingthymeleaf.html#parameterizable-fragment-signatures
Need to pass a generic `Pageable` object to the fragment, not specific to Company.